### ISSUE TITLE
check that reported `search` containment ani is using the right containment direction

### ISSUE THREAD

**Issue by bluegenes** [sourmash-bio/sourmash#2193]
for @ctb -

If we're running `sourmash search`, are we producing the containment of the query in the reference, or the reference in the query? Thinking about it now I think we're doing query in the reference.

Over here https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/search.py#L324, I think we're getting containment ANI relative to query (but using containment we generated during search), which would be wrong.

Generally, I think this is related to the syntax/meanings of `containment_ani` vs `contained_by`.

`self.contained_by(other)` = containment of self in other
`self.containment_ani` = ANI from containment of other in self

I think, besides fixing, it would probably be helpful to standardize the python api -- maybe `contained_by_to_ani`?

I think `prefetch` and `gather` results should be fine, because we generate containment, ani within those, so it was clear which direction to go.

**Comment by ctb** [comment]
> If we're running `sourmash search`, are we producing the containment of the query in the reference, or the reference in the query? Thinking about it now I think we're doing query in the reference.

yes. (I like to think about it as "query in mixture or metagenome", just because "reference" is ambiguous given our various use cases!)

> Over here https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/search.py#L324, I think we're getting containment ANI relative to query (but using containment we generated during search), which would be wrong.
> 
> Generally, I think this is related to the syntax/meanings of `containment_ani` vs `contained_by`.
> 
> `self.contained_by(other)` = containment of self in other `self.containment_ani` = ANI from containment of other in self

yep - `contained_by` resulted from one of the earliest confusions for me! https://github.com/sourmash-bio/sourmash/pull/199
 
> I think, besides fixing, it would probably be helpful to standardize the python api -- maybe `contained_by_to_ani`?

I slightly prefer either `contained_by_as_ani` but that's just a suggestion!

**Comment by ctb** [comment]
the line in search.py is:
```python
            self.cmp.estimate_mh1_containment_ani(containment = self.similarity)
```
should it be `estimate_mh2_containment_ani` instead?
```python
            self.cmp.estimate_mh2_containment_ani(containment = self.similarity)
```

**Comment by bluegenes** [comment]
yep - that's the quick fix!

The slightly larger fix is `containment_ani` --> `contained_by_as_ani` (which I quite like). This would fix `compare` direction as well, #2194

**Comment by bluegenes** [comment]
> yep - contained_by resulted from one of the earliest confusions for me! https://github.com/sourmash-bio/sourmash/pull/199

Also, I can('t) believe I repeated your journey to `contained_by`. I found `contained_by` pretty frustrating and `containment` seemed so clear ... until it really wasn't and `contained_by` was better all along 🤦🏻‍♀️ 😂 .

**Comment by ctb** [comment]
> > yep - contained_by resulted from one of the earliest confusions for me! #199
> 
> Also, I can('t) believe I repeated your journey to `contained_by`. I found `contained_by` pretty frustrating and `containment` seemed so clear ... until it really wasn't and `contained_by` was better all along 🤦🏻‍♀️ 😂 .

sometimes even professors figure stuff out 😜

**Comment by ctb** [comment]
working on it here: https://github.com/sourmash-bio/sourmash/pull/2215

please feel free to take it over :)

**Comment by ctb** [comment]
I ended up with something even more verbose over in #2215 -

* `mh1_containment_ani` became `ani_from_mh1_containment_in_mh2`
* `estimate_mh1_containment_ani` became `estimate_ani_from_mh1_containment_in_mh2`

and the converse for mh2. I think it is pretty readable but I'm blinded by the code changes at this point so hot takes are welcome!

**Comment by ctb** [comment]
FWIW it looks like we were using the right containment direction after all - at least, when I run things by hand and/or encode them in tests, what I get out of the CLI matches my expectations.

See most specifically the `asymmetry` tests for `search` and `prefetch` in #2215, where query-in-merged (containment == 1.0) gives an ANI of 1.0, and merged-in-query (containment < 1) gives an ANI of 0.98651.
